src/skills.ts

```diff
@@ -71,30 +83,77 @@ declare global {
     pos: [x: number, y: number]
   ): Promise<unknown>;
 
+  // cburst with no target
+  // var hostiles=get_nearby_hostiles({range:character.range-2,limit:12}),targets=[],mp=character.mp-200,hmp=parseInt(mp/hostiles.length);
+	// 		hostiles.forEach(function(hostile){
+	// 			targets.push([hostile.id,hmp]);
+	// 		});
+
   /** The string is the ID of the target, the number is how much mana to spend on the attack */
   function use_skill(
     name: "cburst",
     targets: [target: SkillTarget, mana: number][]
   ): Promise<unknown>;
 
-  function use_skill(name: "3shot", targets: Tuple<SkillTarget, 1>): Promise<Tuple<unknown, 1>>;
-  function use_skill(name: "3shot", targets: Tuple<SkillTarget, 2>): Promise<Tuple<unknown, 2>>;
-  function use_skill(name: "3shot", targets: Tuple<SkillTarget, 3>): Promise<Tuple<unknown, 3>>;
-
-  function use_skill(name: "5shot", targets: Tuple<SkillTarget, 1>): Promise<Tuple<unknown, 1>>;
-  function use_skill(name: "5shot", targets: Tuple<SkillTarget, 2>): Promise<Tuple<unknown, 2>>;
-  function use_skill(name: "5shot", targets: Tuple<SkillTarget, 3>): Promise<Tuple<unknown, 3>>;
-  function use_skill(name: "5shot", targets: Tuple<SkillTarget, 4>): Promise<Tuple<unknown, 4>>;
-  function use_skill(name: "5shot", targets: Tuple<SkillTarget, 5>): Promise<Tuple<unknown, 5>>;
+  // if(target && is_array(target))
+	// 	for(var i=0;i<target.length;i++)
+	// 	{
+	// 		if(target[i] && target[i].id) target[i]=target[i].id; // "3shot", "5shot"
+	// 		if(target[i] && target[i][0] && target[i][0].id) target[i][0]=target[i][0].id; // "cburst"
+	// 	}
+
+  /** if no target is supplied, it will find 3 targets in character.range-2 */
```

Comment:
Instead of modifying all overloads, add a new one which doesn't take a `targets` argument and add your comment to this one. (Also probably make it so the returned value is `Promise<unknown>` or `Promise<Array<unknown>>`.

Reply:
Yeah, would probably be prettier and easier to maintain.